src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsQueryHandler.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentValidation;
+using MediatR;
+using SFA.DAS.Events.Domain.Entities;
+using SFA.DAS.Events.Domain.Repositories;
+
+namespace SFA.DAS.Events.Application.Queries.GetAgreementEvents
+{
+    public sealed class GetAgreementEventsQueryHandler : IAsyncRequestHandler<GetAgreementEventsRequest, GetAgreementEventsResponse>
+    {
+        private readonly IAgreementEventRepository _agreementEventRepository;
+
+        public GetAgreementEventsQueryHandler(IAgreementEventRepository agreementEventRepository)
+        {
+            _agreementEventRepository = agreementEventRepository;
+        }
+
+        public async Task<GetAgreementEventsResponse> Handle(GetAgreementEventsRequest request)
+        {
+            Validate(request);
+
+            var events = await _agreementEventRepository.GetByDateRange(request.FromDateTime, request.ToDateTime, request.PageSize, request.PageNumber, request.FromEventId);
+
+            return new GetAgreementEventsResponse { Data = MapFrom(events)};
+        }
+
+        private static IEnumerable<AgreementEvent> MapFrom(IEnumerable<AgreementEvent> source)
```

Comment:
for mapping entities to api types (there's a separate library for api client types)